Substrate front-end tutorial (#191)
substrate-developer-hub/substrate-developer-hub.github.io: docs/tutorials/basic-dapp/part-1-connect-to-a-node.md

    return (
      <Dimmer active>
        <Loader size='small'>{text}</Loader>
      </Dimmer>
    );
  };
  
  if(!apiReady){
    return loader('Connecting to the blockchain')
  }

  return (
    <Container>
      Connected
    </Container>
  );
```
As described before, this code will verify that the `api` from the state is initialized and ready before showing "Connected" to our users
The `loader` function returns a nice looking loader with any text we want. This loader logic has been extracted into its own function because we will reuse it later in the tutorial.

You can get the working version of this code by visiting `part-1-1` directory and launch it with:
```bash
cd part-1-1;
yarn;
yarn start;
```

You should see the loader for a couple of seconds, and then "Connected" 🚀.

## 1.2 Retrieve node information

Now that we are successfully connected to our node let's present some information about it.
We will create a new component called `NodeInfo` responsible for displaying this information, and we will pass the `api` in the props.
All we need to import are the usual `React` component and the hook functions.

```js
import React, {useEffect, useState} from 'react';
```

> The [Polkadot-js api](https://polkadot.js.org/api/api/) allows you to send RPC queries to the node.

Reviewer: ```suggestion
The [Polkadot-js api](https://polkadot.js.org/api/api/) allows you to send RPC calls to the node.
```

> To get the node's information we want, we will use the `rpc.system` method and section given by the `api` promise.

Reviewer: ```suggestion
To get the node's information we want, we will use the `rpc.system` methods provided by the `api`.
```

> We will persist in the state the information returned as an object, to display it later on.

Reviewer: ```suggestion
We will add the information returned by the API to the components state to be able to access and display it later on.
```

```js
export default function NodeInfo(props) {
  const {api} = props;
  const [nodeInfo, setNodeInfo] = useState({})

  useEffect(() => {
    Promise.all([
      api.rpc.system.chain(),
      api.rpc.system.name(),
      api.rpc.system.version(),
    ])
    .then(([chain, nodeName, nodeVersion]) => {
      setNodeInfo ({
        chain,
        nodeName,
        nodeVersion
      })
    })
    .catch((e) => console.error(e));
  },[api.rpc.system]);
```

> Remember what we said in the previous part? The second argument of `useEffect` is an array that presents the variables to watch. In this case, `useEffect` will be triggered when the component loads and whenever `api.rpc.system` changes. It should not but we never know (and if you don't specify this argument the component will work surely as well but you'll see a warning in the console).

Reviewer: ```suggestion
Remember what we said in the previous part? The second argument of `useEffect` is an array that presents the variables to watch. In this case, `useEffect` will be triggered when the component loads and whenever `api.rpc.system` changes.
The API endpoints do not change unless there's a runtime upgrade of the node. Even though this will not happen in our development environment and the component will work without specifying this argument, we are still passing the `api.rpc.system` to `useEffect` to make our examples as realistic as possible.
```

> Finally, we'll display these info at the top of the page:

Reviewer: ```suggestion
Finally, we'll display our node information at the top of the page:
```
